File: src-tauri/src/commands/drive.rs
use tauri::State;

use crate::drive::{self, DriveStatus};
use crate::error::Result;
use crate::state::AppState;

/// Current studio-root reachability snapshot; broadcasts keep it fresh after
/// this initial query.
#[tauri::command]
#[specta::specta]
pub async fn get_drive_status(state: State<'_, AppState>) -> Result<DriveStatus> {
    drive::current(&state.db).await
}

File: src/lib/ipc/drive.ts
import type { DriveStatus } from "@/lib/ipc/bindings.gen";
import { commands } from "@/lib/ipc/bindings.gen";
import { unwrap } from "@/lib/ipc/result";

export type { DriveStatus };

export const driveKeys = {
	status: ["drive", "status"] as const,
};

/** Snapshot of studio-root reachability; broadcasts keep the cache fresh. */
export const getDriveStatus = (): Promise<DriveStatus> =>
	unwrap(commands.getDriveStatus());
